### DISCUSSION TITLE
Commerce URL is giving standard error of "The value of the property 'clientAppInsightsKey' cannot be parsed. The error is: Unrecognized Guid format

### DISCUSSION THREAD

**Discussion by shaheen8** [microsoft/Dynamics365Commerce.Solutions#318] [Q&A]
I am following the standard document of installing self -hosted CSU in dev machine-
[https://learn.microsoft.com/en-us/dynamics365/commerce/dev-itpro/install-csu-dev-env](url)

I did the steps mentioned in the section -"Create channel profiles" and pushed the full sync job. However, when I navigate the Commerce URL I am getting an error: 

![image](https://github.com/user-attachments/assets/ed3efef6-96bf-457f-a5df-495983daab7e)

When I navigate to the file "Microsoft.Dynamics.Retail.RetailServer.AspNetCore.dll.config"mentioned I notice that these values are missing from the config file 

![image](https://github.com/user-attachments/assets/df427bd6-c0b1-43b7-a83f-bd858c027c63)

However, other dev machines where this document was followed have these default values.

I have already done these steps : 
1.	Initialized the commerce scheduler
2.	Generate AX7 queries

**Comment by mkaczmarek-evenica** [comment]
Had this exact same issue and could not find any solutions online. One of our D365 devs was able to figure it out.

-Sign into F&O and search for "Monitoring and Telemetry Parameters"
-On the left, find "Application Insights Registry"
-Copy one of the Insights Keys (I used the one marked "Test")
-Open your config xml for the CSU install and fill in ALL the empty insights key values (ALL of them - even the windows phone one)
-Run the CSU install again

**Reply by PrajualR** [reply]
The application insights registry is empty for my dev env

**Comment by sebacn** [comment]
Had similar issue in D365 dev env,
When run install cmd for: CommerceStoreScaleUnitSetup.exe install --port 446 --Config "[path to xml config file]" --SSLCertThumbprint ...
in the config file (--Config) were missing related data for clientAppInsightsKey,  hardwareStationAppinsightsKey and windowsPhonePosAppInsightsKey
workaround for dev env: added guid keys into config file and run installation again

**Reply by PrajualR** [reply]
Where to get the guid keys
